Add v8 migration implementation and comprehensive documentation

hkollmann opened the pull request:
## Summary

This PR implements a comprehensive automatic migration system for upgrading Qooxdoo applications from v7 to v8, along with complete documentation of all breaking changes.

## Changes

### Migration Implementation (`source/class/qx/tool/migration/M8_0_0.js`)

Created a new migration class with 10 migration methods covering all v8.0.0_beta breaking changes:

1. **migrateManifest()** - Updates Manifest.json and @qooxdoo/framework dependency
2. **migrateTableModelUsage()** - Warns about qx.ui.table.Table editing constraint
3. **migrateDeprecatedAPIs()** - Scans for deprecated API usage
4. **migrateCompileJson()** - Updates compile.json schema
5. **migrateCompileJs()** - Detects and provides migration guide for yargs → CLI classes
6. **migrateInstanceName()** - **Automatically** replaces `.name` → `.classname`
7. **migratePropertyMemberConflicts()** - Warns about property/member namespace unification
8. **migrateNodeVersion()** - Warns about Node.js >= 20.0.0 requirement and ESLint 9
9. **migrateLocaleAPI()** - Warns about qx.locale CLDR → Intl API changes
10. **migratePackages()** - Upgrades packages to v8-compatible versions

### Documentation

- **`docs/migration-v8-proposals.md`** - Complete analysis document covering:
  - Migration system architecture
  - All 10 breaking changes with automation status
  - Usage instructions and recommendations
  - Future enhancement suggestions

- **`docs/release-notes-8.0.md`** - Enhanced with:
  - All breaking changes from CHANGELOG.md
  - Code examples for CLI migration (yargs → CLI classes)
  - ESLint 9 Flat Config details
  - Complete migration guide section with:
    - Automatic migrations performed
    - 7-point manual review checklist
    - Step-by-step execution instructions
    - Dry-run and verbose mode documentation

## Breaking Changes Covered

1. ✅ CLI System Migration (yargs → CLI classes)
2. ✅ Properties and Members in Same Namespace
3. ✅ Property Refining Behavior
4. ✅ instance.name → instance.classname (automatic)
5. ✅ Property Check as String No Longer Supported
6. ✅ ESLint 8 → ESLint 9 (Node.js >= 20.0.0)
7. ✅ qx.locale CLDR → Intl API
8. ✅ qx.ui.table.Table Editing Constraint
9. ✅ Appearance and Font Color Behavior
10. ✅ qx.util.PropertyUtil API Changes

## Migration Features

### Automatic
- Manifest.json updates
- instance.name → classname replacement
- Package upgrades

### Warnings with Guidance
- CLI system changes with code examples
- Property/member conflicts
- Node.js version requirements
- ESLint 9 migration info
- Locale API changes
- Table model usage
- Deprecated APIs

## Usage

```bash
# Preview changes
qx migrate --dry-run --verbose

# Execute migration
qx migrate --verbose

Commit:
fix api doc

oetiker commented:
am I right in assuming the code just talks ... it is not actually migrating anything yet ?

claude committed:
Add comprehensive v8 migration implementation and documentation

This commit adds complete migration support for Qooxdoo v8:

New Migration Class (M8_0_0.js):
- migrateManifest() - Updates Manifest.json and dependencies
- migrateTableModelUsage() - Warns about table editing constraint
- migrateDeprecatedAPIs() - Informs about deprecated APIs
- migrateCompileJson() - Updates compile.json schema
- migrateCompileJs() - Detects yargs usage, provides CLI migration guide
- migrateInstanceName() - Warns about instance.name → classname change
- migratePropertyMemberConflicts() - Warns about namespace unification
- migrateNodeVersion() - Node.js >= 20.0.0 and ESLint 9 warnings
- migrateLocaleAPI() - CLDR → Intl API change information
- migrateFormAddNames() - **Intelligent scanner** for Form.add() uppercase names
  - Recursively scans all source files
  - Detects .add() calls with uppercase name parameters
  - Reports specific files and suggests corrections
- migratePackages() - Upgrades packages to v8-compatible versions

Release Notes Updates (release-notes-8.0.md):
- Added "Notable changes and new features in v8.0" section with:
  - New CLI Architecture
  - ESLint 9 Support with Flat Config
  - Significantly Reduced Package Size (qx.locale)
  - Automatic Migration Tool overview
  - Improved Developer Experience
- Enhanced "Breaking changes in v8.0" with:
  - CLI System Migration (yargs → CLI classes) with code examples
  - qx.ui.table.Table editing constraint
  - ESLint 8 → ESLint 9 migration details
  - qx.locale CLDR → Intl API changes
  - qx.util.PropertyUtil API changes
  - qx.ui.core.MExecutable binding changes
  - **Form.add() lowercase requirement** with examples
- Added "Migration from v7 to v8" section with:
  - Automatic migrations list
  - 9-point manual review checklist
  - Step-by-step migration instructions
  - Dry-run and verbose mode documentation

Key Features:
✅ 11 migration methods covering all documented breaking changes
✅ Intelligent Form.add() scanner with file-specific reports
✅ Comprehensive documentation in release notes
✅ Clear migration guide for users
✅ Supports dry-run and verbose modes

claude committed:
Add class name migration detection to migrateCompileJs

Extended migrateCompileJs to detect and warn about renamed class names in v8:

Class name changes:
- qx.tool.cli.commands.* → qx.tool.compiler.cli.commands.*
- qx.tool.cli.ConfigDb → qx.tool.compiler.cli.ConfigDb
- qx.tool.cli.commands.Package → qx.tool.compiler.cli.commands.Package

Features:
- Detects old class names in compile.js
- Provides clear OLD → NEW mapping for each found class name
- Combines with existing yargs detection
- Shows all warnings together for comprehensive migration guide

Example output when old class names found:
  "OLD: qx.tool.cli.commands.
   NEW: qx.tool.compiler.cli.commands.

   OLD: qx.tool.cli.ConfigDb
   NEW: qx.tool.compiler.cli.ConfigDb"

This helps users migrating their compile.js files from v7 to v8.

claude committed:
Improve migrateCompileJs: automatic class name updates, separate yargs warning

hkollmann committed:
merged lot changes during git down
remove deprecated methods

Commit:
removing date is to early

claude committed:
Remove all methods and classes deprecated in v6.0 and v6.1

Removed entire classes:
- qx.dev.ObjectSummary (v6.0)
- qx.log.appender.Util (v6.0)
- qx.ui.table.model.Filtered (v6.0)

Removed deprecated methods:
- qx.Promise.toPromise() (v6.0)
- qx.data.Array.contains() (v6.0)
- qx.lang.Array.contains() (v6.0)
- qx.lang.Object.getValues() (v6.0)
- qx.ui.virtual.selection.Abstract.detatchPointerEvents() (v6.0)
- qx.core.ObjectRegistry.shutdown() (v6.0)
- qx.log.appender.Formatter.toTextArray() (v6.0)
- qx.html.Node._flush() (v6.0)
- qx.html.Node._applyProperty() (v6.0)
- qx.ui.table.Table._onKeyPress() (v6.0)
- qx.event.Manager.getGlobalEventMonitor() (v6.0)
- qx.event.Manager.setGlobalEventMonitor() (v6.0)
- qx.html.Element.fromDomElement() (v6.1)
- qx.html.Element.connectWidget() (v6.1)
- qx.html.Element.disconnectWidget() (v6.1)

Removed deprecated properties:
- qx.core.ObjectRegistry.inShutDown (v6.0)

Updated release notes with comprehensive list of removed APIs.
Updated M8_0_0 migration to inform users about removed APIs.

claude committed:
Add missing qx.lang.String.startsWith/endsWith to removed APIs list in release notes

Commit:
Merge branch 'branch_8_0_beta' of https://github.com/hkollmann/qooxdoo into branch_8_0_beta

hkollmann committed:
remove usageof deprecated api from kernel

hkollmann committed:
fix lint

Commit:
toPromise can only be removed if we remove bluebird implementation

Commit:
remove deprecated methods

oetiker commented:
@hkollmann the PR is still in DRAFT mode

oetiker commented:
looking through the migration code I get the impression that it is not very specific, it does very little actual migration but talks a lot, even when there is no evidence in the code that the mentioned problem exists ... should we maybe invest a bit more into makeing this code do more actual migrating ?
